### PR TITLE
Documentation for UI features

### PR DESCRIPTION
**Pull request by nearnshaw**
To merge when 6.1 is released

@oscar please keep in mind for translations

### CODE REVIEW

File: _posts/development-guide/2018-02-15-onscreen-ui.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+  - development-guide
+type: Document
+set: development-guide
+set_order: 15
+---
+
+
+There are several special component types that are meant for using in a 2D screen space as part of the UI, instead of in the 3D world space. These components are displayed fixed on the player's screen.
+
+UI elements are only visible when the player is standing inside the scene's LAND parcels, as neighboring scenes might have their own UI to display.  When the player clicks the _close UI_ button, on the top-right corner of the screen, all UI elements go away.
```
Comment (pravusjif) [inline _posts/development-guide/2018-02-15-onscreen-ui.md]: the ui toggle button is actually in the bottom-right corner of the screen ;)

File: _posts/development-guide/2018-02-15-onscreen-ui.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+
+To add a screenspace UI to your scene, you must create an entity and add a `UICanvas` component to it. All the visible UI elements that you want the player to see are added as additional objects that are children of this parent object with a `UICanvas` component.
+
+<!--
+![](/images/media/UI-basic.png)
+-->
+
+```ts
+// Create entity
+const ui = new Entity()
```
Comment (pravusjif) [inline _posts/development-guide/2018-02-15-onscreen-ui.md]: this entity instantiation and its following UI component addition to the entity, and its addition to the engine, is not necessary (we aborted the ui components migration plan that took advantage of entity parenting due to unforeseen complications)

File: _posts/development-guide/2018-02-15-onscreen-ui.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+## Types of UI content
+
+There are several different types of UI elements you can add to the screenspace:
+
+
+- Images: Add a `UIImage` component to display any image. Use the `source` field to point to the path of the image.
+
+- Text: Add a `UIText`component to display text. The properties you can set are the same as in a `TextShape` component.  See [text]({{ site.baseurl }}{% post_url /development-guide/2018-02-11-text %}).
+
+- Buttons: Add a `UIButton` to add a clickable button. The button offers some visual feedback when players mouse over it and when they click it.
```
Comment (pravusjif) [inline _posts/development-guide/2018-02-15-onscreen-ui.md:53]: UIButton hasn't been implemented in unity yet.

File: _posts/development-guide/2018-02-15-onscreen-ui.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+Certain UI elements are there to help you organize how you place other elements.
+
+<!--
+![](/images/media/UI-rectangle.png)
+-->
+
+For this, you can use the `UIContainerStack` and the `UIContainerRect`.
+
+Both these shapes have properties to set their color, line thickness, and rounded corners.
```
Comment (pravusjif) [inline _posts/development-guide/2018-02-15-onscreen-ui.md]: we can't set "rounded corners"

**Review by matiasbargas — APPROVED** [review]

**Pull request merged**